File: Typescript/src/uniqueMorseRepresentations.ts
export function uniqueMorseRepresentations(words: string[]): number {
	const morseCode: string[] = [".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."];
	const transformations: Set<string> = new Set();
	for (let word of words) {
		let transformation = '';
		for (let char of word) {
			transformation += morseCode[char.charCodeAt(0) - 'a'.charCodeAt(0)];
		}
		transformations.add(transformation);
	}
	return transformations.size;
};

File: Dart/lib/uniqueMorseRepresentations.dart
class Solution {
  int uniqueMorseRepresentations(List<String> words) {
    List<String> morseCode = [
      ".-",
      "-...",
      "-.-.",
      "-..",
      ".",
      "..-.",
      "--.",
      "....",
      "..",
      ".---",
      "-.-",
      ".-..",
      "--",
      "-.",
      "---",
      ".--.",
      "--.-",
      ".-.",
      "...",
      "-",
      "..-",
      "...-",
      ".--",
      "-..-",
      "-.--",
      "--.."
    ];

    Set<String> set = {};

    for (String word in words) {
      String code = "";
      for (int i = 0; i < word.length; i++) {
        int index = word.codeUnitAt(i) - 'a'.codeUnitAt(0);
        code += morseCode[index];
      }
      set.add(code);
    }
    return set.length;
  }
}
